spec/ar_query_matchers/queries/load_counter_spec.rb

```diff
@@ -77,4 +77,15 @@
 
     expect(stats.query_counts).to eq('MockUser' => 1, 'MockPost' => 1)
   end
+
+  it 'returns the correct result when there are subqueries' do
+    user = MockUser.create!(name: 'Daniel')
+    MockPost.create!(mock_user: user)
+
+    stats = described_class.instrument do
+      MockPost.where(mock_user: MockUser.where(name: 'Daniel')).to_a
```

Comment:
@iking96, I'm trying to understand the intended purpose of the change. 

This gem provides a matcher that would be used as such:

```ruby
expect { 
  MockPost.where(mock_user: MockUser.where(name: 'Daniel')).to_a 
}.to only_load_models('MockUser' => 1, 'MockPost' => 1)
```
Given this query, I'd expect to only see `MockPost`, the data for `MockUser` is never loaded or transferred over the wire.

Reply:
@mzruya 

The sql version of the instrumented query looks like this:
```
SELECT "mock_posts".* FROM "mock_posts" WHERE "mock_posts"."mock_user_id" IN (SELECT "mock_users"."id" FROM "mock_users" WHERE "mock_users"."name" = ?)
```

**Our assumption:** I from the query above I had thought two queries would happen. So we'd want this helper to return both tables. Though it makes sense that only the final result would come over the wire.

Is this incorrect -- perhaps due to the way the DB optimizes things most of the time?

**Something we noticed along the way:**
In the current code the `MockPost Load` printout would result in `MockPost` be returned, but if the [sql regex](https://github.com/Gusto/ar-query-matchers/blob/c8341a15d053344a7838986e147e5958fb339e48/lib/ar_query_matchers/queries/load_counter.rb#L24) were allowed to evaluate the printout it would match `MockUser` because it finds the outermost match.

Reply:
> Our assumption: I from the query above I had thought two queries would happen. So we'd want this helper to return both tables. Though it makes sense that only the final result would come over the wire.
> Is this incorrect -- perhaps due to the way the DB optimizes things most of the time?

I think it's technically correct, but perhaps inaccurate.
We're sending a single SQL statement over the wire. The AST of a SQL statement is recursive.
How the database decides to execute it–and the complexity of doing so–is completely up to its implementation. 
It seems like the new sql regex is trying to asses the complexity executing a query – this isn't something that I think is possible to do via regular expressions or part of the responsibility of this gem.

> Something we noticed along the way:
> In the current code the MockPost Load printout would result in MockPost be returned, but if the sql regex were allowed to evaluate the printout it would match MockUser because it finds the outermost match.

The `ar-query-matchers` gem operates closer to the ORM layer.
Its intended purpose is to make it clear when inefficient use of the ORM yields multiple sql statements.

In this specific example though, we are loading `MockPost` via a single sql query. 
Similarly, we could have done `MockPost.joins(:mock_user)`. In both cases, the ORM doesn't actually fetch `MockUser`.

Compare this to:
```ruby
  MockPost.includes(:mock_user).to_a 
```

Which would invoke 2 sql queries, 1 to load all the mock posts, and 1 to load all their respective mock users.

Reply:
@mzruya I see, thanks! I reigned in the scope of the change just to address the regex issue mentioned above!